v2.0/training/sql-basics.md

```diff
@@ -0,0 +1,503 @@
+transaction midway if, for example, you discover that the transfer has
+insufficient funds.
+
+Here's the above example in a transaction. Again, don't run this example yet.
+
+~~~ sql
+> BEGIN;
+-- Now adding input for a multi-line SQL transaction client-side.
+-- Press Enter two times to send the SQL text collected so far to the server, or Ctrl+C to cancel.
+-- You can also use \show to display the statements entered so far.
+                        -> SELECT balance >= 250 FROM bank.accounts WHERE type = 'checking' AND customer_id = 2;
+                        -> -- press Enter again
++----------------+
+| balance >= 250 |
++----------------+
+|      true      |
++----------------+
+(1 row)
+
+-- If false, issue a ROLLBACK statement. Otherwise, continue.
+
+OPEN> UPDATE bank.accounts SET balance = balance - 250 WHERE type = 'savings' AND customer_id = 2;
+UPDATE 1;
+OPEN> COMMIT;
+COMMIT
+~~~
+
+Try running two copies of the above transaction in parallel. You'll need to open
+another SQL shell. Then, for each query in the example above, execute it in the
+first shell, then the new shell. That is, execute `BEGIN` in the first shell,
+then again in the new shell, then execute `SELECT balance >= 300...` in the
+first shell, then again in the new shell, and so on.
+
+When you reach the `COMMIT` statement, you'll see one transaction fail with an
+error like this:
+
+~~~
+pq: restart transaction: HandledRetryableTxnError: TransactionRetryError: retry txn (RETRY_WRITE_TOO_OLD)
+~~~
+
+CockroachDB has automatically detected that the two transactions are attempting
+conflicting withdrawals and canceled one of them! Just like that, we've plugged
+the leak in our bank.
+
+{{site.data.alerts.callout_info}}
+Any number of `SELECT`, `INSERT`, `UPDATE`, and `DELETE` queries can be placed
+in a transaction. This is what makes traditional SQL databases so powerful.
+{{site.data.alerts.end}}
+
+## Step 8. Aggregations
+
+`SELECT` statements aren't limited to combining data from different tables. They
+can also combine data in the same table using _aggregations_. For example, we
+can sum the balances in the accounts table to determine the total assets we
+manage:
+
+~~~ sql
+> SELECT SUM(balance) FROM bank.accounts;
++----------+
+|   sum    |
++----------+
+| 42564.15 |
++----------+
+(1 row)
+~~~
+
+Or perhaps we want the balance grouped by account type:
+
+~~~ sql
+> SELECT type, SUM(balance) FROM bank.accounts GROUP BY type;
++----------+----------+
+|   type   |   sum    |
++----------+----------+
+| checking |   250.00 |
+| savings  | 42314.15 |
++----------+----------+
+(2 rows)
+~~~
+
+SQL is even more powerful than we've demonstrated here. Joins and aggregations
+can be combined and nested to express nearly any query you can conceive of. If
+you'd like, [continue learning about CockroachDB
+SQL](../learn-cockroachdb-sql.html) in our docs.
+
+## What's Next?
+
+- [Cluster Startup and Scaling](cluster-startup-and-scaling.html)
```

Comment:
As mentioned above, I think it's important to start with basic operations before going into sql. I'd put this no earlier than Data Import.